dask_expr/concat.py

```diff
@@ -0,0 +1,121 @@
+import functools
+
+import pandas as pd
+from dask.dataframe import methods
+from dask.dataframe.dispatch import make_meta, meta_nonempty
+from dask.dataframe.utils import check_meta, strip_unknown_categories
+from dask.utils import apply, is_dataframe_like, is_series_like
+
+from dask_expr.expr import AsType, Expr
+
+
+class Concat(Expr):
+    _parameters = ["join", "ignore_order", "_kwargs"]
+    _defaults = {"join": "outer", "ignore_order": False, "_kwargs": {}}
+
+    @property
+    def _frame(self):
+        return self.operands[len(self._parameters) :]
+
+    @functools.cached_property
+    def _meta(self):
+        meta = make_meta(
+            methods.concat(
+                [meta_nonempty(df._meta) for df in self._frame],
```

Comment:
I'm curious, why nonempty?  What broke?

Reply:
The dtype inference in pandas is buggy with empty dfs, so would rather not rely on it